plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt

```diff
@@ -151,7 +155,9 @@ class CodeScanSessionConfig(
                     val current = stack.pop()
 
                     if (!current.isDirectory) {
-                        if (!changeListManager.isIgnoredFile(current) && !files.contains(current.path)
+                        if (!changeListManager.isIgnoredFile(current) &&
+                            runBlocking { !featureDevSessionContext.ignoreFile(current, this) } &&
+                            !files.contains(current.path)
```

Comment:
please confirm there is no regression for large project trees

Reply:
still unclear why we can't start converging implementations https://github.com/aws/aws-toolkit-jetbrains/pull/4471

Reply:
We have different implementations for iterating through files(we need changeListManager which is not setup in 4471), different file size limits. So not sure about merging both implementations. But if other teams are ok with using changeListManager along with existing ignore setup, then I can try merging implementations.

Reply:
There is some slow down in file collection time.